### DISCUSSION TITLE
Join Table Engine with ReplaceMergeTree functionality

### DISCUSSION THREAD

**Discussion by rhcgn** [ClickHouse/ClickHouse#33980] [Q&A] [answered]
I have a table hits, which represents all hits and I have another table customers, which represents all known customers.
hits table increases fast and has a MergeTree engine, customer table is more about updating the customer information over time, it has ReplaceMergeTree engine.
Both of them are combined with a materialized view via a join to create visits table, which includes all information about visits and if known also customer information of the visitor like zip code, size or age.
In order to speed up materialized view I would like to use a join table engine on customers table. But now I run into the problem to update customer information, which is currently done via the ReplaceMergeTree engine.
Any idea or best practices? Maybe using cascaded MVs? Thanks for your help.

**Comment by den-crane** [comment, marked as answer]
Engine=JOIN is the incorrect tool here. External dictionaries solve this problem, especially check partial updates in external dictionaries.

Ext.dictionaries support 2 method to access their data, dictGet and Join.

ReplacingMergeTree can be a source of Ext.dictionary (source Clickhouse) https://gist.github.com/den-crane/3cf085dc0e36d9c51bde825aa72a8ae9

https://altinity.com/blog/tag/dictionaries/

https://kb.altinity.com/altinity-kb-dictionaries/partial-updates/

**Reply by rhcgn** [reply]
Thank you so much. Will have a look at it. +1
